Sources/MapboxCoreNavigation/Distance/LineDistanceInfo.swift

```diff
@@ -0,0 +1,42 @@
+import Foundation
+import MapboxNavigationNative
+
+/**
+ * Contains information about distance to the road object represented as line.
+ */
+public struct LineDistanceInfo {
+
+    /**
+     * Distance along the road graph from current position to entry point of object in meters,
+     * if already "within" object will be equal to zero
+     */
+    public let distanceToEntry: CLLocationDistance
+
+    /** Distance along the road graph from current position to exit point of object in meters. */
+    public let distanceToExit: CLLocationDistance
+
+    /** Distance along the road graph from current position to end of road object */
+    public let distanceToEnd: CLLocationDistance
```

Comment:
Perhaps the documentation should clarify the difference between an exit and an end. I don’t think that distinction would be intuitive for developers.

Reply:
@SiarheiFedartsou @etl could you please provide more explanation for the docs?

Reply:
See this conversation from Android repo https://github.com/mapbox/mapbox-navigation-android/pull/4364#discussion_r631165151

Reply:
Documentation clarified as part of #2991.